### PR TITLE
DOC rework persistence user guide

### LINKED ISSUE #220: Docs: Update persistence docs to reflect latest audit changes
The [current persistence docs](https://github.com/skops-dev/skops/blob/ca9302146ce53cd89d73fc85d6ffb8143fa6d877/docs/persistence.rst) still reflect the state from before we added the audit feature. This should be updated soon, since we want people to test the feature and the documentation should be accurate for that.

### PR DESCRIPTION
**Pull request by adrinjalali**
Fixes #220. 

This should better reflect the state of affairs now.

cc @skops-dev/maintainers

### CODE REVIEW

**Comment by adrinjalali** [comment]
Also cc @thomasjpfan. This should address the comments you had.

File: docs/persistence.rst [hunk 1]
```diff
@@ -31,8 +30,12 @@ the `Python docs
     untrusted source, or that could have been tampered with.
 
 In contrast to ``pickle``, the :func:`skops.io.dump` and :func:`skops.io.load`
-functions cannot be used to save arbitrary Python code, but they bypass
-``pickle`` and are thus more secure.
+functions have a more limited scope, while preventing users from running
+arbitrary code or loading unknown and malicious objects.
+
+When loading a file, :func:`skops.io.load`/:func:`skops.io.loads` will read
+traverse the input and check for known and unknown types, and will only
```
Comment (BenjaminBossan) [inline docs/persistence.rst:37]:
```suggestion
traverse the input, check for known and unknown types, and will only
```

Also: "will read traverse": should be "read" or "traverse", not both

File: docs/persistence.rst [hunk 2]
```diff
@@ -42,15 +45,22 @@ The code snippet below illustrates how to use :func:`skops.io.dump` and
 
 .. code:: python
 
-    from sklearn.linear_model import LogisticRegression
+    from xgboost.sklearn import XGBClassifier
```
Comment (BenjaminBossan) [inline docs/persistence.rst:48]: Maybe mention above that users need xgboost to run this?

File: docs/persistence.rst [hunk 3]
```diff
@@ -64,28 +74,35 @@ using :func:`skops.io.get_untrusted_types`:
...
 includes complex use cases such as :class:`sklearn.pipeline.Pipeline`,
-:class:`sklearn.model_selection.GridSearchCV`, classes using Cython code, such
-as :class:`sklearn.tree.DecisionTreeClassifier`, and more. If you discover an
-sklearn estimator that does not work, please open an issue on the skops `GitHub
-page <https://github.com/skops-dev/skops/issues>`_ and let us know.
-
-In contrast to ``pickle``, skops cannot persist arbitrary Python code. This
-means if you have custom functions (say, a custom function to be used with
+:class:`sklearn.model_selection.GridSearchCV`, classes using objects defined in
+Cython such such as :class:`sklearn.tree.DecisionTreeClassifier`, and more. If
```
Comment (BenjaminBossan) [inline docs/persistence.rst:96]:
```suggestion
Cython such as :class:`sklearn.tree.DecisionTreeClassifier`, and more. If
```

File: docs/persistence.rst [hunk 4]
```diff
@@ -96,32 +113,29 @@ most types from **numpy** and **scipy** should be supported, such as (sparse)
...
-
-To follow what features are currently planned, filter for the `"persistence"
-label <https://github.com/skops-dev/skops/labels/persistence>`_ in our GitHub
-issues.
+There needs to be more testing to harden the loader and make sure we don't run
+arbitrary code when it's not intended. However, the safety mechanisms already
+in place should prevent most cases of abuse.
+
+At the moment most persisting and loading arbitrary C extension types is not
+possible unless a python object wraps around them and handles persistance and
```
Comment (BenjaminBossan) [inline docs/persistence.rst:134]:
```suggestion
At the moment, persisting and loading arbitrary C extension types is not
possible, unless a python object wraps around them and handles persistence and
```

More readable IMO

**Review by BenjaminBossan — CHANGES_REQUESTED** [review]
Good update. Only minor issues from my side.

**Review by BenjaminBossan — APPROVED** [review]
Thanks!

**Pull request merged**